### PR TITLE
feat: adding the ability to pass a regex to filter tickets

### PR DESCRIPTION
**Pull request by wovalle**
Hey, nice project!

In my company we have many jira projects and instead of passing a fixed array of prefixes, we needed to pass a regex and get the ticket name on the fly. 

I added some documentation around it.

Thanks!

### CODE REVIEW

**Comment by GabrielCastro** [comment]
Thanks for the contribution

File: lib/success.ts [hunk 1]
```diff
@@ -6,10 +6,16 @@ import { GenerateNotesContext, PluginConfig } from './types';
 import { escapeRegExp } from './util';
 
 function getTickets(config: PluginConfig, context: GenerateNotesContext): string[] {
-  const patterns = config.ticketPrefixes!
+  let patterns: RegExp[] = [];
+
+  if (config.ticketRegex) {
+    patterns = [new RegExp(escapeRegExp(config.ticketRegex), 'giu')];
```
Comment (GabrielCastro) [inline lib/success.ts]: we should remove the `escapeRegExp` call here otherwise it won't match

**Comment by wovalle** [comment]
Hey! Sorry it took me so long to reply, I was on vacations!

I updated the function and added a couple of tests to verify that everything work correctly :)

**Comment by wovalle** [comment]
cc @guigonc

**Comment by GabrielCastro** [comment]
Thanks for the change including setting up tests 💯

**Pull request merged**